toqito/state_props/common_quantum_overlap.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from toqito.state_opt.state_exclusion import state_exclusion
+
+
+
+def common_quantum_overlap(states: list[np.ndarray]) -> float:
+    r"""Calculate the common quantum overlap of a collection of quantum states.
+    
+    The common quantum overlap ω_Q[n] quantifies the "overlap" between n quantum states
+    based on their antidistinguishability properties. It is related to the 
+    antidistinguishability probability A_Q[n] by the formula:
+    
+    ω_Q[n] = n(1 - A_Q[n])
+    
+    
+    For two pure states with inner product |⟨ψ|φ⟩| = p, the common quantum overlap is:
+    ω_Q = (1 - sqrt(1-|p|²))
+    
+    The common quantum overlap is a key concept in analyzing epistemic models of quantum
+    mechanics and understanding quantum state preparation contextuality.
+    
+    Parameters
+    ----------
+    states : list[np.ndarray]
+        A list of quantum states represented as numpy arrays. States can be pure states
+        (represented as state vectors) or mixed states (represented as density matrices).
+        
+    Returns
+    -------
+    float
+        The common quantum overlap value.
```

Comment:
we have a different style for this. Please see the contributing guide in the documentation. 

Also, you don't need to list the type in this section if the parameter is type hinted on L6.